admin-console/src/main/webapp/src/pages/ApplicationPage.tsx

```diff
@@ -79,12 +85,26 @@ export const ApplicationPage: React.FunctionComponent = () => {
         return undefined;
     }, [ aggregationTemplates.payload?.status, aggregationTemplates.payload?.value ]);
 
+    const templates = useMemo(() => {
+        if (getAllTemplates.payload?.status === 200) {
+            return getAllTemplates.payload.value;
+        }
+
+        return undefined;
+    }, [ getAllTemplates.payload?.status, getAllTemplates.payload?.value ]);
+
     const createEventType = () => {
         setShowModal(true);
         setIsEdit(false);
         setEventTypes({});
     };
 
+    const createInstantTemplate = () => {
+        setShowModal(true);
+        setIsEdit(false);
+        setInstantTemplates({});
```

Comment:
Try to have all these variables condensed into a single one

Reply:
like 
`  const [ createEditInstantTemplateModal, setCreateEditInstantTemplateModal ] =
    React.useState({ showModal: false, isEdit: false, instantTemplates: {}});`

Reply:
Somewhat like that sounds good.